pages/[...slug].vue

```diff
@@ -22,18 +22,17 @@
       );
     } catch (err: any) {
       if (err?.name === 'FetchError' && err?.status === 404) {
-        router.replace('/404');
+        window.history.pushState({}, '', '/404');
```

Comment:
Could you please test if `router.push('/404');` does the same that you are trying to do here